zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -174,16 +176,21 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
 
   override final def runAsync[A](fa: RIO[R, A])(
     cb: Either[Throwable, A] => effect.IO[Unit]
-  ): effect.SyncIO[Unit] =
+  ): effect.SyncIO[Unit] = {
+    implicit def trace: ZTraceElement = CoreTracer.newTrace
+
     effect.SyncIO {
       rts.unsafeRunAsyncWith(fa.exit) { exit =>
         cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
       }
     }
+  }
 
   override final def runCancelable[A](fa: RIO[R, A])(
     cb: Either[Throwable, A] => effect.IO[Unit]
-  ): effect.SyncIO[effect.CancelToken[RIO[R, *]]] =
+  ): effect.SyncIO[effect.CancelToken[RIO[R, *]]] = {
+    implicit def trace: ZTraceElement = CoreTracer.newTrace
```

Comment:
Anytime you have a user-defined lambda (including a `Function0` generated with a by-name parameter `=>`), you need to trace the lambda and use the `ZTraceElement` from that lambda.

Reply:
Here you have user-defined callback `cb` which will have a source location attached.

Reply:
fixed